<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function will read in a .bam file, annotate the taxonomy and genome
names, reduce the mapping ambiguity using a mixture model, and output a
.csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file)."><title>MetaScope ID — metascope_id • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MetaScope ID — metascope_id"><meta property="og:description" content="This function will read in a .bam file, annotate the taxonomy and genome
names, reduce the mapping ambiguity using a mixture model, and output a
.csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.99</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/rmd/MetaScope_Vignette.html">Tutorial</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/compbiomed/MetaScope">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>MetaScope ID</h1>
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/MetaScope/blob/HEAD/R/MetaScope_ID.R" class="external-link"><code>R/MetaScope_ID.R</code></a></small>
      <div class="d-none name"><code>metascope_id.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function will read in a .bam file, annotate the taxonomy and genome
names, reduce the mapping ambiguity using a mixture model, and output a
.csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">metascope_id</span><span class="op">(</span>
  <span class="va">bam_file</span>,
  aligner <span class="op">=</span> <span class="st">"subread"</span>,
  NCBI_key <span class="op">=</span> <span class="cn">NULL</span>,
  out_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/fileutils.html" class="external-link">file_path_sans_ext</a></span><span class="op">(</span><span class="va">bam_file</span><span class="op">)</span>, <span class="st">".metascope_id.csv"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>,
  EMconv <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">10000</span>,
  EMmaxIts <span class="op">=</span> <span class="fl">25</span>,
  num_species_plot <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>bam_file</dt>
<dd><p>The .bam file that needs to be summarized, annotated, and
needs removal of ambiguity.</p></dd>
<dt>aligner</dt>
<dd><p>The aligner which was used to create the bam file. Default is 
"subread" but can also be set to "bowtie" or "other"</p></dd>
<dt>NCBI_key</dt>
<dd><p>(character) NCBI Entrez API key. optional.
See taxize::use_entrez(). Due to the high number of
requests made to NCBI, this function will be less prone to errors
if you obtain an NCBI key.
You may enter the string as an input or set it as ENTREZ_KEY in .Renviron.</p></dd>
<dt>out_file</dt>
<dd><p>The name of the .csv output file. Defaults to the bam_file
basename plus ".metascope_id.csv".</p></dd>
<dt>EMconv</dt>
<dd><p>The convergence parameter of the EM algorithm. Default set at
<code>1/10000</code>.</p></dd>
<dt>EMmaxIts</dt>
<dd><p>The maximum number of EM iterations, regardless of whether
the EMconv is below the threshhold. Default set at <code>50</code>.
If set at <code>0</code>, the algorithm skips the EM step and summarizes the .bam
file 'as is'</p></dd>
<dt>num_species_plot</dt>
<dd><p>The number of genome coverage plots to be saved.
Default is <code>NULL</code>, which saves coverage plots for the ten most
highly abundant species.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>This function returns a .csv file with annotated read counts to genomes with
mapped reads. The function itself returns the output .csv file name.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co">#### Align reads to reference library and then apply metascope_id()</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Assuming filtered bam files already exist</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Subread aligned bam file</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Create object with path to filtered subread bam file </span></span>
<span class="r-in"><span class="va">bamPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"subread_target.filtered.bam"</span>,</span>
<span class="r-in">package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Run metascope id with the aligner option set to subread</span></span>
<span class="r-in"><span class="fu">metascope_id</span><span class="op">(</span>bam_file <span class="op">=</span> <span class="va">bamPath</span>, aligner <span class="op">=</span> <span class="st">"subread"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading .bam file: /scratch/3276553.1.linga/Rtmph47wXA/temp_libpath1d6ac7e37a51d/MetaScope/extdata/subread_target.filtered.bam</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 856 reads aligned to 77 NCBI accessions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Obtaining taxonomy and genome names</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 6 unique NCBI taxonomy IDs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting up the EM algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting EM iterations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]   1.0000 812.7488</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 2.0000000 0.8139525</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 3.0000000 0.4173796</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 4.0000000 0.5594038</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 5.0000000 0.6864919</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 6.0000000 0.7528093</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 7.0000000 0.7763587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 8.0000000 0.7836479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 9.0000000 0.7861907</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 10.0000000  0.7873258</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 11.0000000  0.7879545</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 12.0000000  0.7883482</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 13.000000  0.788609</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 14.000000  0.788786</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 15.0000000  0.7889075</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 16.0000000  0.7889914</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 17.0000000  0.7890495</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 18.00000  0.78909</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 19.0000000  0.7891182</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 20.0000000  0.7891379</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 21.0000000  0.7891516</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 22.0000000  0.7891613</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 23.0000000  0.7891681</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 24.0000000  0.7891729</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 25.0000000  0.7891762</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	DONE! Converged in 25 interations.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found reads for 5 genomes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating coverage plots</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Results written to /scratch/3276553.1.linga/Rtmph47wXA/temp_libpath1d6ac7e37a51d/MetaScope/extdata/subread_target.filtered.metascope_id.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TaxonomyID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     273036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2     418127</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3     523796</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4     426430</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5     158878</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                          Genome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                                                Staphylococcus aureus RF122, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                        Staphylococcus aureus subsp. aureus Mu3, whole genome shotgun sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 Staphylococcus aureus subsp. aureus ST398 strain 55488 NODE_10, whole genome shotgun sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                            Staphylococcus aureus subsp. aureus str. Newman, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                                   Staphylococcus aureus subsp. aureus Mu50, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   read_count          Proportion EMreads        EMProportion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        844   0.985981308411215     844   0.985968411207406</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2          6 0.00700934579439252     3.8 0.00440144211459077</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3          2 0.00233644859813084     2.3 0.00271337559673324</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4          2 0.00233644859813084     2.2 0.00257358934601975</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5          2 0.00233644859813084     3.7 0.00434318173524827</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Bowtie aligned bam file </span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Create object with path to filtered subread bam file </span></span>
<span class="r-in"><span class="va">bamPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"bowtie_target.filtered.bam"</span>,</span>
<span class="r-in">package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Run metascope id with the aligner option set to bowtie</span></span>
<span class="r-in"><span class="fu">metascope_id</span><span class="op">(</span>bam_file <span class="op">=</span> <span class="va">bamPath</span>, aligner <span class="op">=</span> <span class="st">"bowtie"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading .bam file: /scratch/3276553.1.linga/Rtmph47wXA/temp_libpath1d6ac7e37a51d/MetaScope/extdata/bowtie_target.filtered.bam</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 955 reads aligned to 120 NCBI accessions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Obtaining taxonomy and genome names</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 6 unique NCBI taxonomy IDs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting up the EM algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting EM iterations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]   1.0000 884.5066</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 2.0000000 0.9827828</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 3.0000000 0.7919072</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 4.0000000 0.7871586</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 5.0000000 0.8068483</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 6.0000000 0.8193158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 7.0000000 0.8275384</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 8.0000000 0.8332668</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 9.0000000 0.8372379</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 10.000000  0.839936</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 11.0000000  0.8417388</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 12.0000000  0.8429307</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 13.0000000  0.8437146</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 14.0000000  0.8442325</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 15.0000000  0.8445838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 16.0000000  0.8448411</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 17.0000000  0.8450599</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 18.0000000  0.8452807</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 19.0000000  0.8455173</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 20.0000000  0.8457242</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 21.0000000  0.8457509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 22.0000000  0.8453414</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 23.0000000  0.8442443</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 24.000000  0.842381</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 25.0000000  0.8398935</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	DONE! Converged in 25 interations.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found reads for 3 genomes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating coverage plots</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Results written to /scratch/3276553.1.linga/Rtmph47wXA/temp_libpath1d6ac7e37a51d/MetaScope/extdata/bowtie_target.filtered.metascope_id.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TaxonomyID                                                             Genome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     273036                     Staphylococcus aureus RF122, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3     158878        Staphylococcus aureus subsp. aureus Mu50, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6     426430 Staphylococcus aureus subsp. aureus str. Newman, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   read_count          Proportion EMreads         EMProportion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        953   0.997905759162304   953.1    0.998026139867055</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3          1 0.00104712041884817     1.3  0.00131623459197387</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6          1 0.00104712041884817     0.6 0.000625302300008886</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Different or unknown aligned bam file</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Create object with path to unknown origin bam file </span></span>
<span class="r-in"><span class="va">bamPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"subread_target.filtered.bam"</span>,</span>
<span class="r-in">package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Run metascope id with the aligner option set to other </span></span>
<span class="r-in"><span class="fu">metascope_id</span><span class="op">(</span>bam_file <span class="op">=</span> <span class="va">bamPath</span>, aligner <span class="op">=</span> <span class="st">"other"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading .bam file: /scratch/3276553.1.linga/Rtmph47wXA/temp_libpath1d6ac7e37a51d/MetaScope/extdata/subread_target.filtered.bam</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 856 reads aligned to 77 NCBI accessions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Obtaining taxonomy and genome names</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 6 unique NCBI taxonomy IDs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting up the EM algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting EM iterations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]   1.0000 759.5111</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 2.0000000 0.9132601</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 3.0000000 0.4755713</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 4.000000 0.568628</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 5.0000000 0.6922415</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 6.0000000 0.7547038</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 7.0000000 0.7766409</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 8.0000000 0.7835835</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 9.0000000 0.7861197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 10.000000  0.787304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 11.0000000  0.7879779</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 12.0000000  0.7884051</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 13.0000000  0.7886894</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 14.000000  0.788883</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 15.0000000  0.7890162</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 16.0000000  0.7891085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 17.0000000  0.7891726</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 18.0000000  0.7892174</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 19.0000000  0.7892487</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 20.0000000  0.7892707</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 21.0000000  0.7892861</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 22.000000  0.789297</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 23.0000000  0.7893047</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 24.0000000  0.7893102</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 25.0000000  0.7893141</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	DONE! Converged in 25 interations.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found reads for 5 genomes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating coverage plots</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving 6.67 x 6.67 in image</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Results written to /scratch/3276553.1.linga/Rtmph47wXA/temp_libpath1d6ac7e37a51d/MetaScope/extdata/subread_target.filtered.metascope_id.csv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TaxonomyID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     273036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2     418127</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3     523796</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4     426430</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5     158878</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                          Genome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                                                Staphylococcus aureus RF122, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                        Staphylococcus aureus subsp. aureus Mu3, whole genome shotgun sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 Staphylococcus aureus subsp. aureus ST398 strain 55488 NODE_10, whole genome shotgun sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                            Staphylococcus aureus subsp. aureus str. Newman, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5                                   Staphylococcus aureus subsp. aureus Mu50, complete sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   read_count          Proportion EMreads        EMProportion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        844   0.985981308411215     844   0.985960405472934</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2          6 0.00700934579439252     3.8 0.00440897491933567</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3          2 0.00233644859813084     2.3 0.00271433413968925</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4          2 0.00233644859813084     2.2 0.00257414474820706</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5          2 0.00233644859813084     3.7 0.00434214071983583</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by W. Evan Johnson, Aubrey R. Odom-Mabey, Rahul Varki.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

